src/views/addTodo.js

```diff
@@ -0,0 +1,131 @@
+import {
+  StyleSheet,
+  Text,
+  View,
+  TextInput,
+  Platform,
+  Button,
+  ScrollView,
+} from 'react-native';
+import React, {useState} from 'react';
+import DateTimePicker from '@react-native-community/datetimepicker';
+import {dateToDDMMYYY} from '../utils/utils';
+import ScreenTitle from '../components/screenTitle';
+
+export const AddTodo = ({navigation}) => {
+  const [title, setTitle] = useState('');
+  const [validTitle, setValidTitle] = useState(true);
+  const [description, setDescription] = useState('');
+  const [validDescription, setValidDescription] = useState(true);
+  const [date, setDate] = useState(new Date());
+  const [show, setShow] = useState(false);
+
+  const onChangeDate = (event, selectedDate) => {
+    const currentDate = selectedDate || date;
+    setShow(Platform.OS === 'ios');
+    setDate(currentDate);
+  };
+
+  const showDatepicker = () => {
+    setShow(true);
+  };
+
+  const checkTodo = () => {
+    let bool = true;
+    if (title.length === 0) {
+      setValidTitle(false);
+      bool = false;
+    }
+
+    if (description.length === 0) {
+      setValidDescription(false);
+      bool = false;
+    }
+
+    if (bool) {
+      navigation.navigate('Home', {
+        new_todo: {
+          id: '100', // todo
+          title: title,
+          description: description,
+          date: date.getTime(),
+          completed: false,
+        },
+      });
+    }
+  };
+
+  return (
+    <ScrollView>
+      <View style={styles.container}>
+        <ScreenTitle>Add Todo</ScreenTitle>
+
+        <View style={styles.inputGroup}>
+          <Text>Title:</Text>
+          <TextInput
+            style={[!validTitle && styles.fieldInvalid]}
```

Comment:
Normalmente usamos variáveis positivas para representar o estado de erro porque acaba por ser mais facil de ler, por exemplo:
```suggestion
            style={[invalidTitle && styles.fieldInvalid]}
```
```suggestion
            style={[titleError && styles.fieldInvalid]}
```